File: Support.Services/Implementations/ContactsService.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text.RegularExpressions;
using Support.Cqrs.Contracts.Contact;
using Support.Cqrs.Queries.Contact;
using Support.Crm.Model.Entities;
using Support.Services.Interfaces;
using Support.Crm.Repository.Interfaces;

namespace Support.Services.Implementations
{
    public class ContactsService : IContactsService
    {
        private readonly IContactsQueries mContactsQueries;
        private readonly Regex mPhoneStartRegex = new Regex(@"^(8|\+7|7)");

        public ContactsService(IContactsQueries contactsQueries)
        {
            mContactsQueries = contactsQueries;
        }

        public ContactResponse GetContactByPhoneNumber(string phoneNumber)
        {
            var normalizedPhone = mPhoneStartRegex.Replace(phoneNumber, "");
            return mContactsQueries.SelectContactsByPhone(normalizedPhone).FirstOrDefault();
        }

        public IEnumerable<ContactResponse> GetContactsBySearchContract(ContactRequest contactSearch)
        {
            return mContactsQueries.SelectContactByRequestParameters(contactSearch);
        }
    }
}

File: Support.CallCenter.Ui/Frontend/Services/ContactsService.js
var ContactsService = function($http)
{
    var getContactByPhoneNumber = function(phoneNumber)
    {
        var requestUrl = "/contacts/" + phoneNumber;

        var requestPromise = $http.get(requestUrl);

        return requestPromise;
    }

    var getContactsBySearchContract = function(contactSearch)
    {
        var requestUrl = "/contacts?FirstName=" + contactSearch.FirstName + "&LastName=" + contactSearch.LastName + "&RouteNumber=" + contactSearch.RouteNumber;

        requestUrl = encodeURI(requestUrl);
        var requestPromise = $http.get(requestUrl);

        return requestPromise;
    }

    var contactsService =
        {
            GetContactByPhoneNumber: getContactByPhoneNumber,
            GetContactsBySearchContract: getContactsBySearchContract
        };
    return contactsService;
}